### PR TITLE
Popup entry with data

### PR DESCRIPTION
**Pull request by hanleymark**
- Removed getEntries.js
- Added getEntries.ts - now has two client side helper functions available to the app:
    - getAllEntries() :IEntry[] Fetch all entries from DB
    - getEntry(id:  number): IEntry Fetch entry with matching id
- Added types folder in src
- Added IEntry interface for passing entry data around
- Minor changes to Entry component to display data passed in as prop
- Changes to page.tsx in /life-in-colour to handle clicks and store last square clicked on in a prop

### CODE REVIEW

**Review by cameochoquer — APPROVED** [review]
looks good. I see another potential for a ternary but can update in my next PR :)

File: src/app/life-in-colour/page.tsx [hunk 1]
```diff
@@ -110,10 +114,17 @@ const Grid: FC = (): ReactElement => {
...
-      {/*isOpen && <Entry onClose={closeModal} entries={entriesData} />*/}
-      <Navbar />
+      {isOpen && <Entry onClose={closeModal} entry={entryClicked} />}
     </>
   );
 };
 
+const getEntryById = (id: number, entries: IEntry[]): IEntry | null => {
+  const entry = entries.find((entry) => entry.id === id);
+  if (entry) {
```
Comment (FomasTreeman) [inline src/app/life-in-colour/page.tsx:124]: entry ? entry : null

File: src/components/Entry.tsx [hunk 2]
```diff
@@ -1,30 +1,19 @@
...
-  entries: UserEntryTable | null;
+  entry: IEntry | null;
 }
 
 //may need to changeReact.FC but version 5.1 should be fine now?
-const Entry: FC<ModalProps> = ({ onClose, entries }): ReactElement => {
-  console.log(`entries from Entry.tsx: ${entries}`);
-  if (!entries) {
+const Entry: FC<ModalProps> = ({ onClose, entry }): ReactElement => {
+  if (!entry) {
```
Comment (FomasTreeman) [inline src/components/Entry.tsx:16]: if (!entry) (....)

File: src/lib/getEntries.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  if (error) {
+    console.error("Error fetching user entries:", error);
+    throw error;
+  }
+  return data;
+}
+
+export async function getEntry(id: number): Promise<IEntry> {
+  const {
```
Comment (FomasTreeman) [inline src/lib/getEntries.ts:26]: theres a function for this in /browser/session

File: src/lib/getEntries.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    .from("entries")
+    .select(
+      "id, mood, mood_date, journal_entry, context_people, context_location"
+    )
+    .eq("user_id", user?.id)
+    .eq("id", id);
+
+  if (error) {
+    console.error("Error fetching user entries:", error);
+    throw error;
```
Comment (FomasTreeman) [inline src/lib/getEntries.ts:42]: do we need to throw?

File: src/components/Entry.tsx
Comment (FomasTreeman) [inline src/components/Entry.tsx:1]: maybe should name this EntryModal

**Review by FomasTreeman — CHANGES_REQUESTED** [review]
small changes that might help with readability.
an extra thing that might be useful is a function in utils for getting the correct date format for when communicating with supabase

File: src/lib/getEntries.ts (hunk 3 above)
Comment (FomasTreeman) [reply src/lib/getEntries.ts:26]: both getAll and getUser entries functions could use this function

**Pull request merged**